controller/Controller.py

```diff
@@ -0,0 +1,207 @@
+
+    def init_dds(self, dds_domain):
+        """Initialize DDS entities."""
+        factory = fastdds.DomainParticipantFactory.get_instance()
+        participant_qos = fastdds.DomainParticipantQos()
+        factory.get_default_participant_qos(participant_qos)
+        participant_qos.name('DDSRecorderController')
+        self.participant = factory.create_participant(
+            dds_domain, participant_qos)
+
+        # Initialize command topic
+        self.command_topic_data_type = ControllerCommandPubSubType()
+        self.command_topic_data_type.setName('ControllerCommand')
+        type_support = fastdds.TypeSupport(self.command_topic_data_type)
+        self.participant.register_type(type_support)
+
+        topic_qos = fastdds.TopicQos()
+        self.participant.get_default_topic_qos(topic_qos)
+        self.command_topic = self.participant.create_topic(
+            'ddsrecorder/controller',
+            self.command_topic_data_type.getName(),
+            topic_qos)
+
+        # Initialize status topic
+        self.status_topic_data_type = StatusPubSubType()
+        self.status_topic_data_type.setName('Status')
+        type_support = fastdds.TypeSupport(self.status_topic_data_type)
+        self.participant.register_type(type_support)
+
+        topic_qos = fastdds.TopicQos()
+        self.participant.get_default_topic_qos(topic_qos)
+        self.status_topic = self.participant.create_topic(
+            'ddsrecorder/status',
+            self.status_topic_data_type.getName(),
+            topic_qos)
+
+        publisher_qos = fastdds.PublisherQos()
+        self.participant.get_default_publisher_qos(publisher_qos)
+        self.publisher = self.participant.create_publisher(publisher_qos)
+
+        self.command_writer_listener = CommandWriterListener(self)
+        writer_qos = fastdds.DataWriterQos()
+        self.publisher.get_default_datawriter_qos(writer_qos)
+        writer_qos.reliability().kind = fastdds.RELIABLE_RELIABILITY_QOS
+        writer_qos.durability().kind = fastdds.TRANSIENT_LOCAL_DURABILITY_QOS
+        writer_qos.history().kind = fastdds.KEEP_LAST_HISTORY_QOS
+        writer_qos.history().size = 10
+        self.command_writer = self.publisher.create_datawriter(
+            self.command_topic, writer_qos, self.command_writer_listener)
+
+        subscriber_qos = fastdds.SubscriberQos()
+        self.participant.get_default_subscriber_qos(subscriber_qos)
+        self.subscriber = self.participant.create_subscriber(subscriber_qos)
+
+        self.status_reader_listener = StatusReaderListener(self)
+        reader_qos = fastdds.DataReaderQos()
+        self.subscriber.get_default_datareader_qos(reader_qos)
+        reader_qos.reliability().kind = fastdds.RELIABLE_RELIABILITY_QOS
+        reader_qos.durability().kind = fastdds.TRANSIENT_LOCAL_DURABILITY_QOS
+        reader_qos.history().kind = fastdds.KEEP_LAST_HISTORY_QOS
+        reader_qos.history().size = 1
```

Comment:
If the info in the external status may change something in the controller, it may be larger than 1.
If it is for debug/user info purpose, it is ok

Reply:
@rsanchez15